parsers/MD.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+"""Parser for Moldova."""
+
+import arrow
+from operator import itemgetter
+import requests
+
+TYPE_MAPPING = {
+    u'tmva476': 'hydro',            # NHE Costeşti (run-of-river) #2 index
+    u'tmva112': 'hydro',      		# NHE Dubăsari (run-of-river) #4 index
+    u'tmva367': 'gas',      		# CET Nord (CHPP) #3 index
+    u'tmva42': 'gas',             	# CET-1 Chişinău (CHPP) #6 index
+    u'tmva378': 'gas',             	# CET-2 Chişinău (CHPP) #5 index
+    u'tmva1024': 'unknown',         # CERS Moldovenească (fuel mix coal, gas, oil) #7 index
+}
+
+display_url = 'http://www.moldelectrica.md/ro/activity/system_state'
+data_url = 'http://www.moldelectrica.md/utils/load4'
+
+def get_data(session = None):
+    """ Returns generation data as a list of floats."""
+
+    s = session or requests.Session()
+
+    #In order for the data url to return data, cookies from the display url must be obtained then reused.
+    response = s.get(display_url)
+    data_response = s.get(data_url)
+    raw_data = data_response.text
+
+    data = [float(i) for i in raw_data.split(',')]
+
+    return data
+
+
+def fetch_production(country_code = 'MD', session = None):
+    """Requests the last known production mix (in MW) of a given country
+
+    Arguments:
+    country_code (optional) -- used in case a parser is able to fetch multiple countries
+    session (optional)      -- request session passed in order to re-use an existing session
+
+    Return:
+    A dictionary in the form:
+    {
+      'countryCode': 'FR',
+      'datetime': '2017-01-01T00:00:00Z',
+      'production': {
+          'biomass': 0.0,
+          'coal': 0.0,
+          'gas': 0.0,
+          'hydro': 0.0,
+          'nuclear': null,
+          'oil': 0.0,
+          'solar': 0.0,
+          'wind': 0.0,
+          'geothermal': 0.0,
+          'unknown': 0.0
```

Comment:
All entries where numbers are set to 0 represent a **known** zero. Are you sure e.g. there's no wind production? Else, better put `None` everywhere and let the parser fill in the blanks.

Reply:
That's in the docstring not the actual parser dict.  I have set wind and solar to None further on.

Reply:
wow sorry my bad!